datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -2763,6 +2790,52 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_array_replace_with_null() {
```

Comment:
Since the null cases were not covered, I added new coverage. Maybe this style of test could help as we clean up the other aray functions

Reply:
I would prefer having test in slt file if possible

Reply:
slt is a good idea -- I will move the tests

Reply:
Done in https://github.com/apache/arrow-datafusion/pull/8054/commits/c6dabcf402c03a0ca602dc72f8cecd8c4e48d1b8